### PR TITLE
Add test for wild fields

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: tests/compile-fail/fields.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#[deny(unneeded_field_pattern)]
+fn main() {
+    let f = Foo { a: 0, b: 0, c: 0 };
+
+    match f {
+        Foo { a: _, b: 0, .. } => {} //~ERROR You matched a field with a wildcard pattern
+                                     //~^ HELP Try with `Foo { b: 0, .. }`
+        Foo { a: _, b: _, c: _ } => {} //~ERROR All the struct fields are matched to a
+                                       //~^ HELP Try with `Foo { .. }`
+    }
```
Comment (Manishearth) [inline tests/compile-fail/fields.rs]: There should be some tests for false positives as well

File: tests/compile-fail/fields.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#![feature(plugin)]
```
Comment (Manishearth) [inline tests/compile-fail/fields.rs]: Probably give the file the same name as the lint

File: tests/compile-fail/fields.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+#[allow(dead_code, unused)]
+
+struct Foo {
+    a: i32,
+    b: i32,
+    c: i32,
+}
+
+#[deny(unneeded_field_pattern)]
```
Comment (Manishearth) [inline tests/compile-fail/fields.rs]: Make this a global deny on the file so it's easy to add new tests

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**